src/app/activitydialog.cpp

```diff
@@ -42,6 +42,8 @@ void ActivityDialog::setActivity(const Activity& activity)
 Activity ActivityDialog::activity() const
 {
     Activity activity;
+    activity.id = 0;
```

Comment:
ID should be left empty or initialized on structure declaration

Reply:
Not sure what you mean exactly. The bug I was hitting was ID would get random values when uninitialized and it would cause API errors when creating new activities.